File: db4on/test/src/com/db4o/db4ounit/Db4oUnitPlatform.cs
using System;
using System.Reflection;
using com.db4o.reflect;
using com.db4o.reflect.generic;

namespace com.db4o.db4ounit
{
	class Db4oUnitPlatform
	{
		public static ReflectClass GetReflectClass(Reflector reflector, Type clazz)
		{
			return reflector.ForClass(j4o.lang.Class.GetClassForType(clazz));
		}

	    public static bool IsStoreableField(FieldInfo a_field)
	    {
	        if (a_field.IsStatic) return false;
            if (j4o.lang.reflect.Field.IsTransient(a_field)) return false;
	        if (a_field.Name.IndexOf("$") != -1) return false;
	        return true;
	    }
	}
}

File: db4oj.tests/src/com/db4o/db4ounit/Db4oUnitPlatform.java
/* Copyright (C) 2004 - 2006 db4objects Inc. http://www.db4o.com */

package com.db4o.db4ounit;

import java.lang.reflect.*;

/**
 * Platform dependent code goes here.
 *
 * @sharpen.ignore
 */
public class Db4oUnitPlatform {

	public static boolean storeableField(Field a_field) {
		return isStoreableField(a_field);
	}

	public static boolean isStoreableField(Field a_field) {
	    return (!Modifier.isStatic(a_field.getModifiers()))
	        && (!Modifier.isTransient(a_field.getModifiers())
	            & !(a_field.getName().indexOf("$") > -1));
	}

}
